File: src/models/pilotOnline.ts
import { Document, model, Schema } from 'mongoose';

interface IPilotOnline extends Document {
	cid: number;
	name: string;
	callsign: string;
	aircraft: string;
	dep?: string;
	dest?: string;
	code?: string;
	lat: number;
	lng: number;
	altitude: number;
	heading: number;
	speed: number;
	planned_cruise: string;
	route?: string;
	remarks?: string;
}

const PilotOnlineSchema = new Schema<IPilotOnline>(
	{
		cid: { type: Number, required: true },
		name: { type: String, required: true },
		callsign: { type: String, required: true },
		aircraft: { type: String, required: true },
		dep: { type: String },
		dest: { type: String },
		code: { type: String },
		lat: { type: Number, required: true },
		lng: { type: Number, required: true },
		altitude: { type: Number, required: true },
		heading: { type: Number, required: true },
		speed: { type: Number, required: true },
		planned_cruise: { type: String },
		route: { type: String },
		remarks: { type: String },
	},
	{ collection: 'pilotsOnline' },
);

export const PilotOnlineModel = model<IPilotOnline>('PilotOnline', PilotOnlineSchema);

File: internal/models/pilotOnline.go
package models

import "go.mongodb.org/mongo-driver/v2/bson"

type PilotOnline struct {
	ID            bson.ObjectID `bson:"_id,omitempty"`
	CID           int           `bson:"cid"`
	Name          string        `bson:"name"`
	Callsign      string        `bson:"callsign"`
	AircraftType  string        `bson:"aircraft"`
	Departure     *string       `bson:"dep"`
	Destination   *string       `bson:"dest"`
	Code          *string       `bson:"code"`
	Latitude      float64       `bson:"lat"`
	Longitude     float64       `bson:"lng"`
	Altitude      int           `bson:"altitude"`
	Heading       int           `bson:"heading"`
	Speed         int           `bson:"speed"`
	PlannedCruise string        `bson:"planned_cruise"`
	Route         *string       `bson:"route"`
	Remarks       *string       `bson:"remarks"`
}
